instance_control/command/upgrade.py

```diff
@@ -0,0 +1,68 @@
+import base64
+import logging
+
+import boto3
+from instance_control import config
+from instance_control import node
+from instance_control import piu
+from instance_control import volume
+
+from instance_control.aws import ec2_node
+from instance_control.command import Command
+
+_LOG = logging.getLogger('bubuku.cluster.command.upgrade')
+
+
+class UpgradeCommand(Command):
+    def __init__(self, cluster_name: str, cluster_config_path: str, image_version: str, ip: str, user: str, odd: str):
+        super().__init__(cluster_name, cluster_config_path)
+        self.image_version = image_version
+        self.ip = ip
+        self.user = user
+        self.odd = odd
+
+    def init(self):
+        if self.image_version:
+            self.cluster_config['image_version'] = self.image_version
+        config.validate_config(self.cluster_name, self.cluster_config)
+
+    def start(self):
+        ec2_client = boto3.client('ec2', region_name=self.cluster_config['region'])
+        ec2_resource = boto3.resource('ec2', region_name=self.cluster_config['region'])
+
+        instance = node.get_instance_by_ip(ec2_resource, self.cluster_name, self.ip)
+        # check_current_image_version(instance, self.cluster_config['image_version'])
+
+        piu.stop_taupage(self.ip, self.user, self.odd)
+
+        _LOG.info('Searching for instance %s volumes', instance.instance_id)
+        volumes = ec2_client.describe_instance_attribute(InstanceId=instance.instance_id,
+                                                         Attribute='blockDeviceMapping')
+        data_volume = next(v for v in volumes['BlockDeviceMappings'] if v['DeviceName'] == '/dev/xvdk')
+        data_volume_id = data_volume['Ebs']['VolumeId']
```

Comment:
Maybe it makes sense to call piu.stop_taupage() right after finding this volume, cause otherwise it will stuck in the middle of upgrade process